packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -162,8 +162,14 @@ module.exports = async function watchmanCrawl(
         if (isOld) {
           files[name] = existingFileData;
         } else {
+          let sha1hex = fileData['content.sha1hex'];
+
+          if (typeof sha1hex !== 'string' && sha1hex.length === 40) {
```

Comment:
Is this if statement correct? Shouldn’t it check if its not a string or a length other than 40? This looks to me like you are unsetting the sha1 when it’s correct. Jetlag?

Reply:
You are right! Glad to see tests caught it.